src/utils/converter.ts

```diff
@@ -27,3 +28,13 @@ export function convertResultToResource(result: WithId<Document>): Resource {
 		}),
 	};
 }
+
+export function convertResultToQuickLink(result: WithId<Document>): QuickLink {
+	return {
+		title: result.title ?? "",
+		description: result.description ?? "",
+		imgURL: result.imgURI ? `${FILE_ENDPOINT_URL}/${result.imgURI}` : null,
+		link: result.link ?? "",
+		originalLink: result.originalLink ?? "",
+	};
```

Comment:
This assumes that result will always be an object, and that pretty much all of them are optional.. It'll then be up to the consumer to check every property xd

Reply:
Actually there's a validation when inserting new data on admin panel, so it's very unlikely that these fields will be missing. Personally I'd say the frontend doesn't need to check again.

Though, is there a better way to do this?

Reply:
If there's already validation, this should throw if the data is not supposed to be empty, but is empty

Reply:
something like this?
```js 
if (!result.title || !result.description || !result.imgURI || !result.link || !result.originalLink) {
    throw new Error("Missing required fields");
  }

return {
	title: result.title,
	description: result.description,
	imgURL: `${FILE_ENDPOINT_URL}/${result.imgURI}`,
	link: result.link,
	originalLink: result.originalLink,
};

Reply:
> something like this?
> 
> ```js
> if (!result.title || !result.description || !result.imgURI || !result.link || !result.originalLink) {
>     throw new Error("Missing required fields for Quicklink object");
>   }
> 
> return {
> 	title: result.title,
> 	description: result.description,
> 	imgURL: `${FILE_ENDPOINT_URL}/${result.imgURI}`,
> 	link: result.link,
> 	originalLink: result.originalLink,
> };
> ```

Sure, or something like

```typescript
const missingField = ['title', 'description', 'imgURI', 'link', 'originalLink'].find(field => !result[field]);

if (missingField) {
    throw new Error(`Missing required field: ${missingField} for Quicklink object`);
}
```

Reply:
Ye that looks cleaner, I'll also apply this to other converterFunc too

Reply:
come to think of it...

if the error is thrown out, it'll be caught by the try catch block in the `resources.controller.ts` right - so, with the current implementation, the server will return an internal server error even when a single result is invalid

it might be a little weird right? Instead, what if we log it and then just exclude it from the resulting array instead?

Reply:
pushed what I meant now, can you have a look :3

Reply:
Good enough for me XD